tests/pass-dep/libc/libc-fs.rs

```diff
@@ -380,6 +382,25 @@ fn test_sync_file_range() {
     assert_eq!(result_2, 0);
 }
 
+#[cfg(target_os = "linux")]
+fn test_fstat() {
+    use std::mem::MaybeUninit;
+    use std::os::unix::io::AsRawFd;
+
+    let path = utils::prepare_with_content("miri_test_libc_fstat.txt", b"hello");
+    let file = File::open(&path).unwrap();
+    let fd = file.as_raw_fd();
+
+    let mut stat: libc::stat = unsafe { MaybeUninit::zeroed().assume_init() };
+    let res = unsafe { libc::fstat(fd, &mut stat) };
+    assert_eq!(res, 0);
+
+    assert_eq!(stat.st_size, 5);
+    assert_eq!(stat.st_mode & libc::S_IFMT, libc::S_IFREG);
```

Comment:
The test should also read all of the other fields that POSIX mandates, to make sure we have initialized them: https://pubs.opengroup.org/onlinepubs/009695299/basedefs/sys/stat.h.html.

And on systems that have `st_mtime_nsec` and friends, those should be read as well. This seems to be Linux, FreeBSD, macOS, Solarish... so, everything we test.